pkg/transforms/vap.go

```diff
@@ -0,0 +1,225 @@
+		Name:      v.node.Properties["name"].(string),
+		NameSpace: "_NONE",
+		UID:       v.node.UID,
+		EdgeType:  "attachedTo",
+		Kind:      v.node.Properties["kind"].(string),
+	}
+
+	propSet := map[string]struct{}{policyName: {}}
+
+	edges := edgesByDestinationName(propSet, "ValidatingAdmissionPolicy", nodeInfo, ns, []string{})
+
+	if v.paramRef == nil {
+		return edges
+	}
+
+	policy, ok := ns.ByKindNamespaceName["ValidatingAdmissionPolicy"]["_NONE"][policyName]
+	if !ok {
+		return edges
+	}
+
+	paramKind, ok := policy.Properties["paramKind_kind"].(string)
+	if !ok || paramKind == "" {
+		return edges
+	}
+
+	paramApiVersion, ok := policy.Properties["paramKind_apiVersion"].(string)
+	if !ok || paramApiVersion == "" {
+		return edges
+	}
+
+	paramGV, err := schema.ParseGroupVersion(paramApiVersion)
+	if err != nil {
+		return edges
+	}
+
+	namespaceToName := ns.ByKindNamespaceName[paramKind]
+
+	if v.paramRef.Namespace != "" {
+		// If the namespace is specified, then limit the searches to just this namespace
+		namespaceToName = map[string]map[string]Node{
+			v.paramRef.Namespace: ns.ByKindNamespaceName[paramKind][v.paramRef.Namespace],
+		}
+	}
+
+	var paramRefSelector labels.Selector
+
+	if v.paramRef.Selector != nil {
+		var err error
+
+		paramRefSelector, err = metav1.LabelSelectorAsSelector(v.paramRef.Selector)
+		if err != nil {
+			return edges
+		}
+	}
+
+	for _, nodeMap := range namespaceToName {
+		for _, node := range nodeMap {
+			version, ok := node.Properties["apiversion"].(string)
+			if !ok || version != paramGV.Version {
+				continue
+			}
+
+			group, ok := node.Properties["apigroup"].(string)
+			if (!ok && group != "") || group != paramGV.Group {
+				continue
+			}
+
+			if v.paramRef.Name != "" && v.paramRef.Name != node.Properties["name"].(string) {
+				continue
+			}
+
+			if paramRefSelector != nil {
+				nodeLabels, ok := node.Properties["label"].(map[string]string)
+				if !ok {
+					continue
+				}
+
+				nodeLabelsSet := labels.Set(nodeLabels)
+
+				if !paramRefSelector.Matches(nodeLabelsSet) {
+					continue
+				}
+			}
+
+			edges = append(edges, Edge{
+				EdgeType:   "paramReferences",
+				SourceUID:  v.node.UID,
+				SourceKind: "ValidatingAdmissionPolicyBinding",
+				DestKind:   paramKind,
+				DestUID:    node.UID,
+			})
+		}
+	}
+
+	return edges
+}
+
+type VapResource struct {
+	node Node
+}
+
+// Validating admission policy
+// Ref: https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/
+func VapResourceBuilder(v *unstructured.Unstructured) *VapResource {
+	node := transformCommon(v)
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       v.GetKind(),
+		APIVersion: v.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node)
+
+	node.Properties["paramKind_kind"], _, _ = unstructured.NestedString(v.Object, "spec", "paramKind", "kind")
+	node.Properties["paramKind_apiVersion"], _, _ = unstructured.NestedString(v.Object, "spec", "paramKind", "apiVersion")
```

Comment:
If there's no special logic, I prefer to use the genericResourceConfig.
https://github.com/stolostron/search-collector/blob/main/pkg/transforms/genericResourceConfig.go#L32